bot/exts/valentines/markov_poem_generator.py

```diff
@@ -0,0 +1,512 @@
+
+    @command_wrapper
+    async def _get_generated_markov_poem(
+        self,
+        ctx: commands.Context,
+        scheme: str,
+        unit_count: Dict[str, int],
+        time_start: datetime.time,
+        is_first_error: bool
+    ) -> List[str]:
+        """
+        Generates a love poem with the markov chain and sends it off.
+
+        If the count of a unit is one, that means it is the last unit in the
+        rhyme scheme being processed.
+
+        The function will iterate through the units of scheme. If it is a new
+        unit, it will be added to `rhyme_track` as a key with the rhyme set
+        as the value. If a unit is found again, then then the last word of the
+        new line's rhyme set will be added to `rhyme_track`. This is so that
+        all the rhymes are not contingent on the first unit of the scheme.
+        """
+        lines = []
+        rhyme_track = {}  # Maps units to their accumulative rhyme sets
+
+        for unit in scheme:
+            # `last_line` means last line of a rhyme group
+            is_last_line = True if unit_count[unit] == 1 else False
+
+            # Create new stanza
+            if unit == "/":
+                lines.append("")
+                continue
+
+            # Creating a line for the unit
+            if unit not in rhyme_track:
+                new_line = await self._init_unit(
+                    rhyme_track, unit, is_last_line
+                )
+                lines.append(new_line)
+            else:
+                new_line = await self._get_rhyming_line(
+                    word_rhymes=rhyme_track[unit],
+                    existing_lines=lines,
+                    error_info=(ctx, is_first_error, scheme, unit_count,
+                                time_start)
+                )
+                lines.append(new_line)
+
+                # If last line, it will not be referred to again
+                if not is_last_line:
+                    rhyme_track[unit] |= await self._get_rhyme_set(
+                        instance=self,
+                        word=self._get_last_word(new_line)
+                    )
+
+            unit_count[unit] -= 1
+
+        return lines
+
+    async def send_markov_poem(
+        self,
+        ctx: commands.Context,
+        scheme: str,
+        unit_count: Dict[str, int],
+        time_start: datetime.time,
+        is_first_error: bool = True
+    ) -> None:
+        """Gives the user a paginated embed of their markov poem."""
+        lines = await self._get_generated_markov_poem(
+            instance=self,
+            timeout=self.POEM_TIMEOUT,
+            ctx=ctx,
+            scheme=scheme,
+            unit_count=unit_count,
+            time_start=time_start,
+            is_first_error=is_first_error
+        )
+
+        elapsed_time = datetime.now() - time_start
+        elapsed_time = elapsed_time.seconds
+
+        poem_embed = Embed(
+            title="A Markov Poem For " + str(ctx.author.name),
```

Comment:
Replace with f-string
```suggestion
            title=f"A Markov Poem For {ctx.author.name}",
```